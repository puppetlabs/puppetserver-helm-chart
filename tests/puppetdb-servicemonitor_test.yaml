suite: test puppetdb serviceMonitor creation
templates:
  - puppetdb-servicemonitor.yaml
release:
  name: puppetserver
  namespace: puppet
tests:
  - it: should not create the serviceMonitor by default
    asserts:
      - hasDocuments:
          count: 0
  - it: should not create the serviceMonitor if "puppetdb.enabled" is false
    set:
      puppetdb.enabled: false
    asserts:
      - hasDocuments:
          count: 0
  - it: should not create the serviceMonitor if "metrics.prometheus.enabled" is false
    set:
      metrics.prometheus.enabled: false
    asserts:
      - hasDocuments:
          count: 0
  - it: should failed is monitoring.coreos.com/v1 api is not available
    set:
      puppetdb.enabled: true
      metrics.prometheus.enabled: true
    asserts:
      - failedTemplate:
          errorMessage: "You have to deploy monitoring.coreos.com/v1 first"
  - it: should create the ressource with the right name 1/2
    capabilities:
      apiVersions:
        - monitoring.coreos.com/v1
    release:
      name: puppetserver
      namespace: puppet
    set:
      puppetdb.enabled: true
      metrics.prometheus.enabled: true
    asserts:
      - isKind:
          of: ServiceMonitor
      - equal:
          path: metadata.name
          value: puppetserver-puppetdb
  - it: should create the ressource with the right name 2/2
    capabilities:
      apiVersions:
        - monitoring.coreos.com/v1
    release:
      name: puppet
      namespace: puppet
    set:
      puppetdb.enabled: true
      metrics.prometheus.enabled: true
    asserts:
      - isKind:
          of: ServiceMonitor
      - equal:
          path: metadata.name
          value: puppet-puppetserver-puppetdb
  - it: should create the service monitor if "puppetdb.enabled" is true AND "metrics.prometheus.enabled" is true
    capabilities:
      apiVersions:
        - monitoring.coreos.com/v1
    release:
      name: puppetserver
      namespace: puppet
    set:
      puppetdb.enabled: true
      metrics.prometheus.enabled: true
    asserts:
    - isKind:
        of: ServiceMonitor
  - it: should create the service monitor even if monitoring.coreos.com/v1 api is not available if "metrics.prometheus.disableAPICheck" is true
    release:
      name: puppetserver
      namespace: puppet
    set:
      puppetdb.enabled: true
      metrics.prometheus.enabled: true
      metrics.prometheus.disableAPICheck: true
    asserts:
    - isKind:
        of: ServiceMonitor
