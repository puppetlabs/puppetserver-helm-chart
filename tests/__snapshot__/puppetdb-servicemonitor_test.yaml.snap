manifest should match snapshot:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      labels:
        app.kubernetes.io/component: puppetdb
        app.kubernetes.io/instance: puppetserver
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: puppetserver
        app.kubernetes.io/version: 7.17.0
        helm.sh/chart: puppetserver-9.4.0
        release: kube-prometheus-stack
      name: puppetserver-puppetdb
      namespace: puppet
    spec:
      endpoints:
        - honorLabels: true
          interval: 30s
          path: /metrics
          port: metrics
      jobLabel: puppetdb
      namespaceSelector:
        matchNames:
          - puppet
      selector:
        matchLabels:
          app.kubernetes.io/component: puppetdb
          app.kubernetes.io/name: puppetserver
